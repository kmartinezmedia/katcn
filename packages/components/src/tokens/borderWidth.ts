import { type BorderWidthConfig } from '../types/tokens';

const compactBorderWidths: BorderWidthConfig = {
  none: 0,
  thin: 1,
  medium: 2,
  thick: 2,
};

const normalBorderWidths: BorderWidthConfig = {
  none: 0,
  thin: 1,
  medium: 2,
  thick: 4,
};

export { compactBorderWidths, normalBorderWidths };
