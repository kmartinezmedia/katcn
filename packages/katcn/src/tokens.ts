import type {
  AvatarSizeConfig,
  BorderRadiusConfig,
  BorderWidthConfig,
  FontFamilyGlobalConfig,
  FontWeightDescriptive,
  FontWeightNumeric,
  HuesChromaConfig,
  HuesConfig,
  HuesLightnessConfig,
  IconSizeConfig,
  PaletteConfig,
  SpacingConfig,
  TypographyConfig,
  UniversalTokensConfig,
  ZIndexConfig,
} from './types';

const hues: HuesConfig = {
  magenta: 331.6,
  pink: 4.6,
  rose: 7.4,
  red: 27.4,
  sunset: 33.9,
  orange: 42.7,
  nude: 42.5,
  brown: 68.8,
  yellow: 82.9,
  citron: 109.8,
  lime: 133.2,
  green: 154.4,
  mint: 164.9,
  teal: 194.8,
  cyan: 224.1,
  blue: 261,
  indigo: 273.3,
  purple: 289,
  gray: 248.1,
  carbon: 0,
};

const huesLightness: HuesLightnessConfig = {
  '0': '99%',
  '1': '95%',
  '2': '88%',
  '3': '80%',
  '4': '74%',
  '5': '68%',
  '6': '63%',
  '7': '58%',
  '8': '53%',
  '9': '49%',
  '10': '42%',
  '11': '35%',
  '12': '27%',
  '13': '20%',
  '14': '14%',
  '15': '11%',
};

const huesChroma: HuesChromaConfig = {
  '0': 0.03,
  '1': 0.06,
  '2': 0.12,
  '3': 0.14,
  '4': 0.16,
  '5': 0.19,
  '6': 0.2,
  '7': 0.21,
  '8': 0.2,
  '9': 0.19,
  '10': 0.17,
  '11': 0.15,
  '12': 0.12,
  '13': 0.09,
  '14': 0.07,
  '15': 0.05,
};

const palette: PaletteConfig = {
  core: {
    brand: { hue: 'purple', step: 9 },
    accent: { hue: 'blue', step: 9 },
    alert: { hue: 'red', step: 9 },
    positive: { hue: 'green', step: 9 },
    warning: { hue: 'orange', step: 9 },
  },
  background: {
    primary: { hue: 'gray', step: 0 },
    secondary: { hue: 'gray', step: 2 },
  },
  foreground: {
    primary: { hue: 'gray', step: 13 },
    secondary: { hue: 'gray', step: 14 },
    tertiary: { hue: 'gray', step: 9 },
    'on-color': { hue: 'gray', step: 0 },
  },
  line: {
    primary: { hue: 'gray', step: 14 },
    secondary: { hue: 'gray', step: 9 },
  },
};

const sansFallbacks = ['Helvetica Neue', 'Helvetica', 'Arial', 'sans-serif'];
const serifFallbacks = ['Georgia', 'Times', 'Times New Roman', 'serif'];

const fontFamily: FontFamilyGlobalConfig = {
  icons: { fallbacks: ['icons', ...sansFallbacks], name: 'icons' },
  sans: { fallbacks: sansFallbacks, name: 'sans' },
  'sans-condensed': { fallbacks: sansFallbacks, name: 'sans-condensed' },
  'serif-text': { fallbacks: serifFallbacks, name: 'serif-text' },
  'serif-display': { fallbacks: serifFallbacks, name: 'serif-display' },
  mono: {
    fallbacks: ['Menlo', 'Monaco', 'Courier New', 'monospace'],
    name: 'mono',
  },
};

const typography: TypographyConfig = {
  display1: {
    fontFamily: 'sans',
    fontSize: '48px',
    lineHeight: '44px',
    fontWeight: 'black',
    textTransform: 'uppercase',
  },
  display2: {
    fontFamily: 'sans',
    fontSize: '40px',
    lineHeight: '44px',
    fontWeight: 'extrabold',
  },
  display3: {
    fontFamily: 'sans',
    fontSize: '36px',
    lineHeight: '44px',
    fontWeight: 'extrabold',
  },
  title1: {
    fontFamily: 'sans',
    fontSize: '32px',
    lineHeight: '40px',
    fontWeight: 'bold',
  },
  title2: {
    fontFamily: 'sans',
    fontSize: '28px',
    lineHeight: '36px',
    fontWeight: 'bold',
  },
  title3: {
    fontFamily: 'sans',
    fontSize: '24px',
    lineHeight: '32px',
    fontWeight: 'bold',
  },
  title4: {
    fontFamily: 'sans',
    fontSize: '20px',
    lineHeight: '24px',
    fontWeight: 'bold',
  },
  headline1: {
    fontFamily: 'sans',
    fontSize: '16px',
    lineHeight: '20px',
    fontWeight: 'semibold',
  },
  body1: {
    fontFamily: 'sans',
    fontSize: '16px',
    lineHeight: '20px',
  },
  label1: {
    fontFamily: 'sans',
    fontSize: '14px',
    lineHeight: '20px',
    fontWeight: 'semibold',
  },
  label2: {
    fontFamily: 'sans',
    fontSize: '14px',
    lineHeight: '20px',
  },
  caption1: {
    fontFamily: 'sans',
    fontSize: '12px',
    lineHeight: '16px',
    fontWeight: 'semibold',
  },
  caption2: {
    fontFamily: 'sans',
    fontSize: '12px',
    lineHeight: '16px',
  },
  legal1: {
    fontFamily: 'sans',
    fontSize: '11px',
    lineHeight: '16px',
    fontWeight: 'semibold',
  },
};

const borderRadius: BorderRadiusConfig = {
  none: '0px',
  xs: '2px',
  sm: '4px',
  md: '8px',
  lg: '16px',
  xl: '32px',
  full: '9999px',
};

const borderWidth: BorderWidthConfig = {
  none: '0px',
  thin: '1px',
  medium: '2px',
  thick: '4px',
};

const spacing: SpacingConfig = {
  px: '1px',
  '0': '0px',
  '0.5': '0.125rem',
  '1': '0.25rem',
  '1.5': '0.375rem',
  '2': '0.5rem',
  '2.5': '0.625rem',
  '3': '0.75rem',
  '3.5': '0.875rem',
  '4': '1rem',
  '5': '1.25rem',
  '6': '1.5rem',
  '7': '1.75rem',
  '8': '2rem',
  '9': '2.25rem',
  '10': '2.5rem',
  '11': '2.75rem',
  '12': '3rem',
  '14': '3.5rem',
  '16': '4rem',
  '20': '5rem',
  '24': '6rem',
  '28': '7rem',
  '32': '8rem',
  '36': '9rem',
  '40': '10rem',
  '44': '11rem',
  '48': '12rem',
  '52': '13rem',
  '56': '14rem',
  '60': '15rem',
  '64': '16rem',
  '72': '18rem',
  '80': '20rem',
  '96': '24rem',
};

const avatarSizes: AvatarSizeConfig = {
  sm: '16px',
  md: '24px',
  lg: '32px',
};

const iconSizes: IconSizeConfig = {
  sm: '16px',
  md: '24px',
  lg: '32px',
};

const zIndex: ZIndexConfig = {
  auto: 'auto',
  '0': '0',
  '10': '10',
  '20': '20',
  '30': '30',
  '40': '40',
  '50': '50',
};

export const defaultTokensConfig: UniversalTokensConfig = {
  palette,
  hues,
  huesChroma,
  huesLightness,
  borderRadius,
  borderWidth,
  spacing,
  avatarSizes,
  iconSizes,
  zIndex,
  fontFamily,
  typography,
};

export const alwaysPalette = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
};

export const fontWeightMap: Record<FontWeightDescriptive, FontWeightNumeric> = {
  thin: '100',
  extralight: '200',
  light: '300',
  regular: '400',
  medium: '500',
  semibold: '600',
  bold: '700',
  extrabold: '800',
  black: '900',
};
