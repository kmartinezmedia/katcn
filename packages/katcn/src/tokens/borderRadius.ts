import type { BorderRadiusConfig } from '../types';

const compactBorderRadius: BorderRadiusConfig = {
  none: 0,
  xs: 2,
  sm: 4,
  md: 8,
  lg: 12,
  xl: 24,
  full: 9999,
};

const normalBorderRadius: BorderRadiusConfig = {
  none: 0,
  xs: 2,
  sm: 4,
  md: 8,
  lg: 16,
  xl: 32,
  full: 9999,
};

export { compactBorderRadius, normalBorderRadius };
