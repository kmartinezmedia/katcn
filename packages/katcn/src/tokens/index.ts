export * from './borderRadius';
export * from './borderWidth';
export * from './defaultTokensConfig';
export * from './elevation';
export * from './layout';
export * from './motion';
export * from './palette';
export * from './sizes';
export * from './spacing';
export * from './spectrum';
export * from './typography';
