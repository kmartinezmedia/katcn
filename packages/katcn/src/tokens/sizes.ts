import type { AvatarSizeConfig, IconSizeConfig } from '../types';

const compactAvatarSizes: AvatarSizeConfig = {
  sm: 16,
  md: 24,
  lg: 32,
};

const normalAvatarSizes: AvatarSizeConfig = compactAvatarSizes;

const compactIconSizes: IconSizeConfig = {
  sm: 16,
  md: 24,
  lg: 32,
};

const normalIconSizes: IconSizeConfig = compactIconSizes;

export {
  compactAvatarSizes,
  compactIconSizes,
  normalAvatarSizes,
  normalIconSizes,
};
