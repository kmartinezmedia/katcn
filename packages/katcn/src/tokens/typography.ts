import type {
  FontFamilyConfig,
  FontFamilyGlobalConfig,
  FontSizeConfig,
  FontWeightConfig,
  FontWeightDescriptive,
  FontWeightNumeric,
  LineHeightConfig,
  TextTransformConfig,
} from '../types';

const sansFallbacks = ['Helvetica Neue', 'Helvetica', 'Arial', 'sans-serif'];
const serifFallbacks = ['Georgia', 'Times', 'Times New Roman', 'serif'];

export const fontFamilyGlobal: FontFamilyGlobalConfig = {
  icons: { fallbacks: ['icons', ...sansFallbacks], name: 'icons' },
  sans: { fallbacks: sansFallbacks, name: 'sans' },
  'sans-condensed': { fallbacks: sansFallbacks, name: 'sans-condensed' },
  'serif-text': { fallbacks: serifFallbacks, name: 'serif-text' },
  'serif-display': { fallbacks: serifFallbacks, name: 'serif-display' },
  mono: {
    fallbacks: ['Menlo', 'Monaco', 'Courier New', 'monospace'],
    name: 'mono',
  },
};

export const fontFamily: FontFamilyConfig = {
  display1: 'sans',
  display2: 'sans',
  display3: 'sans',
  title1: 'sans',
  title2: 'sans',
  title3: 'sans',
  title4: 'sans',
  headline1: 'sans',
  body1: 'sans',
  label1: 'sans',
  label2: 'sans',
  caption1: 'sans',
  caption2: 'sans',
  legal1: 'sans',
};

export const fontWeightMap: Record<FontWeightDescriptive, FontWeightNumeric> = {
  thin: '100',
  extralight: '200',
  light: '300',
  regular: '400',
  medium: '500',
  semibold: '600',
  bold: '700',
  extrabold: '800',
  black: '900',
};

export const fontWeight: FontWeightConfig = {
  display1: 'black',
  display2: 'extrabold',
  display3: 'extrabold',
  title1: 'bold',
  title2: 'bold',
  title3: 'bold',
  title4: 'bold',
  headline1: 'semibold',
  body1: 'regular',
  label1: 'semibold',
  label2: 'regular',
  caption1: 'semibold',
  caption2: 'regular',
  legal1: 'semibold',
};

export const textTransform: TextTransformConfig = {
  display1: 'uppercase',
  display2: 'none',
  display3: 'none',
  title1: 'none',
  title2: 'none',
  title3: 'none',
  title4: 'none',
  headline1: 'none',
  body1: 'none',
  label1: 'none',
  label2: 'none',
  caption1: 'none',
  caption2: 'none',
  legal1: 'none',
};

export const xSmall = {
  fontSize: {
    display1: 46,
    display2: 37,
    display3: 33,
    title1: 29,
    title2: 25,
    title3: 21,
    title4: 17,
    headline1: 13,
    body1: 13,
    label1: 12,
    label2: 12,
    caption1: 11,
    caption2: 11,
    legal1: 10,
  } satisfies FontSizeConfig,
  lineHeight: {
    display1: 44,
    display2: 44,
    display3: 44,
    title1: 36,
    title2: 32,
    title3: 28,
    title4: 24,
    headline1: 16,
    body1: 16,
    label1: 16,
    label2: 16,
    caption1: 16,
    caption2: 16,
    legal1: 12,
  } satisfies LineHeightConfig,
};

export const small = {
  fontSize: {
    display1: 46,
    display2: 38,
    display3: 34,
    title1: 30,
    title2: 26,
    title3: 22,
    title4: 18,
    headline1: 14,
    body1: 14,
    label1: 13,
    label2: 13,
    caption1: 12,
    caption2: 12,
    legal1: 11,
  } satisfies FontSizeConfig,
  lineHeight: {
    ...xSmall.lineHeight,
    title1: 40,
    headline1: 20,
    body1: 20,
    legal1: 16,
  } satisfies LineHeightConfig,
};

export const medium = {
  fontSize: {
    display1: 47,
    display2: 39,
    display3: 35,
    title1: 31,
    title2: 27,
    title3: 23,
    title4: 19,
    headline1: 15,
    body1: 15,
    label1: 13,
    label2: 13,
    caption1: 12,
    caption2: 12,
    legal1: 11,
  } satisfies FontSizeConfig,
  lineHeight: {
    ...small.lineHeight,
    title2: 36,
    title3: 32,
  } satisfies LineHeightConfig,
};

export const large = {
  fontSize: {
    ...medium.fontSize,
    display1: 48,
    display2: 40,
    display3: 36,
    title1: 32,
    title2: 28,
    title3: 24,
    title4: 20,
    headline1: 16,
    body1: 16,
    label1: 14,
    label2: 14,
  } satisfies FontSizeConfig,
  lineHeight: {
    ...medium.lineHeight,
    title2: 36,
    title3: 32,
    headline1: 20,
    body1: 20,
    label1: 20,
    label2: 20,
  } satisfies LineHeightConfig,
};

export const xLarge = {
  fontSize: {
    display1: 50,
    display2: 42,
    display3: 38,
    title1: 34,
    title2: 30,
    title3: 26,
    title4: 22,
    headline1: 18,
    body1: 16,
    label1: 16,
    label2: 16,
    caption1: 14,
    caption2: 14,
    legal1: 13,
  } satisfies FontSizeConfig,
  lineHeight: {
    ...large.lineHeight,
    title1: 44,
    headline1: 24,
    body1: 24,
    caption1: 20,
    caption2: 20,
  } satisfies LineHeightConfig,
};

export const xxLarge = {
  fontSize: {
    display1: 52,
    display2: 44,
    display3: 40,
    title1: 36,
    title2: 32,
    title3: 28,
    title4: 26,
    headline1: 20,
    body1: 20,
    label1: 18,
    label2: 18,
    caption1: 16,
    caption2: 16,
    legal1: 15,
  } satisfies FontSizeConfig,
  lineHeight: {
    ...xLarge.lineHeight,
    title2: 40,
    title3: 36,
    headline1: 28,
    body1: 28,
    label1: 24,
    label2: 24,
    legal1: 20,
  } satisfies LineHeightConfig,
};

export const xxxLarge = {
  fontSize: {
    display1: 54,
    display2: 46,
    display3: 42,
    title1: 38,
    title2: 34,
    title3: 30,
    title4: 28,
    headline1: 22,
    body1: 22,
    label1: 20,
    label2: 20,
    caption1: 18,
    caption2: 18,
    legal1: 17,
  } satisfies FontSizeConfig,
  lineHeight: {
    ...xxLarge.lineHeight,
    title1: 48,
    title2: 44,
    title4: 32,
    label1: 28,
    label2: 28,
    caption1: 24,
    caption2: 24,
  } satisfies LineHeightConfig,
};
