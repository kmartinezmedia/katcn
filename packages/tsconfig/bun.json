{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Bun",
  "extends": "./node.json",
  "compilerOptions": {
    "types": ["bun-types"]
  }
}
