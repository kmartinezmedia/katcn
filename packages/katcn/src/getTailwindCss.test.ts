import { describe, expect, it } from 'bun:test';
import { getTailwindCss } from './getTailwindCss';
import { defaultTokensConfig } from './tokens';

describe('getTailwindCss', () => {
  it('getTailwindCss should return correct vars', () => {
    const css = getTailwindCss(defaultTokensConfig);
    expect(css).toBeDefined();
  });
});
