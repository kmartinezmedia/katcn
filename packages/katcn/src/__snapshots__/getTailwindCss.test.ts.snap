// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`getTailwindCss getTailwindCss should return correct vars 1`] = `
"@import "tailwindcss";
@custom-variant dark (&:where(.dark, .dark *));
@theme {--font-icons: icons;--font-sans: sans;--font-serif: serif;--font-mono: mono;--text-display1: 6rem;--leading-display1: 1.5;--font-weight-display1: 500;--font-display1: var(--font-sans);--text-display2: 3.5rem;--leading-display2: 1.5;--font-weight-display2: 500;--font-display2: var(--font-sans);--text-display3: 2.5rem;--leading-display3: 1.5;--font-weight-display3: 450;--font-display3: var(--font-sans);--text-display4: 2.5rem;--leading-display4: 1.5;--font-weight-display4: 400;--font-display4: var(--font-sans);--text-title1: 1.75rem;--leading-title1: 1.5;--font-weight-title1: 700;--font-title1: var(--font-sans);--text-title2: 1.75rem;--leading-title2: 1.5;--font-weight-title2: 450;--font-title2: var(--font-sans);--text-title3: 1.375rem;--leading-title3: 1.5;--font-weight-title3: 700;--font-title3: var(--font-sans);--text-title4: 1.375;--leading-title4: 1.5;--font-weight-title4: 450;--font-title4: var(--font-sans);--text-headline: 1rem;--leading-headline: 1.25;--font-weight-headline: 500;--font-headline: var(--font-sans);--text-body: 1rem;--leading-body: 1.5;--font-weight-body: 400;--font-body: var(--font-sans);--text-label1: 0.875rem;--leading-label1: 1.5;--font-weight-label1: 500;--font-label1: var(--font-sans);--text-label2: 0.875rem;--leading-label2: 1.5;--font-weight-label2: 400;--font-label2: var(--font-sans);--text-label3: 0.75rem;--leading-label3: 1.5;--font-weight-label3: 500;--font-label3: var(--font-sans);--text-label4: 0.75rem;--leading-label4: 1.5;--font-weight-label4: 400;--font-label4: var(--font-sans);--text-legal: 0.625rem;--leading-legal: 1.5;--font-weight-legal: 400;--font-legal: var(--font-sans);--color-brand: var(--color-blue-500);--color-on-brand: var(--color-zinc-50);--color-accent: var(--color-rose-600);--color-on-accent: var(--color-zinc-50);--color-alert: var(--color-red-500);--color-on-alert: var(--color-zinc-50);--color-positive: var(--color-green-500);--color-on-positive: var(--color-zinc-50);--color-warning: var(--color-orange-500);--color-on-warning: var(--color-zinc-50);--color-primary: var(--color-zinc-50);--color-on-primary: var(--color-zinc-900);--color-secondary: var(--color-zinc-200);--color-on-secondary: var(--color-zinc-600);--color-tertiary: var(--color-zinc-100);--color-on-tertiary: var(--color-zinc-500);--color-inverse: var(--color-zinc-900);--color-on-inverse: var(--color-zinc-50);
}
.dark {--color-on-brand: var(--color-zinc-950);--color-accent: var(--color-rose-400);--color-on-accent: var(--color-zinc-950);--color-on-alert: var(--color-zinc-950);--color-on-positive: var(--color-zinc-950);--color-on-warning: var(--color-zinc-950);--color-primary: var(--color-zinc-950);--color-on-primary: var(--color-zinc-100);--color-secondary: var(--color-zinc-800);--color-on-secondary: var(--color-zinc-400);--color-tertiary: var(--color-zinc-900);--color-inverse: var(--color-zinc-100);--color-on-inverse: var(--color-zinc-950);
}
@utility text-variant-display1 {
	      line-height: var(--line-height-display1);
	      font-size: var(--font-size-display1);
	      font-weight: var(--font-weight-display1);
	      font-family: var(--font-sans);
	    }
@utility text-variant-display2 {
	      line-height: var(--line-height-display2);
	      font-size: var(--font-size-display2);
	      font-weight: var(--font-weight-display2);
	      font-family: var(--font-sans);
	    }
@utility text-variant-display3 {
	      line-height: var(--line-height-display3);
	      font-size: var(--font-size-display3);
	      font-weight: var(--font-weight-display3);
	      font-family: var(--font-sans);
	    }
@utility text-variant-display4 {
	      line-height: var(--line-height-display4);
	      font-size: var(--font-size-display4);
	      font-weight: var(--font-weight-display4);
	      font-family: var(--font-sans);
	    }
@utility text-variant-title1 {
	      line-height: var(--line-height-title1);
	      font-size: var(--font-size-title1);
	      font-weight: var(--font-weight-title1);
	      font-family: var(--font-sans);
	    }
@utility text-variant-title2 {
	      line-height: var(--line-height-title2);
	      font-size: var(--font-size-title2);
	      font-weight: var(--font-weight-title2);
	      font-family: var(--font-sans);
	    }
@utility text-variant-title3 {
	      line-height: var(--line-height-title3);
	      font-size: var(--font-size-title3);
	      font-weight: var(--font-weight-title3);
	      font-family: var(--font-sans);
	    }
@utility text-variant-title4 {
	      line-height: var(--line-height-title4);
	      font-size: var(--font-size-title4);
	      font-weight: var(--font-weight-title4);
	      font-family: var(--font-sans);
	    }
@utility text-variant-headline {
	      line-height: var(--line-height-headline);
	      font-size: var(--font-size-headline);
	      font-weight: var(--font-weight-headline);
	      font-family: var(--font-sans);
	    }
@utility text-variant-body {
	      line-height: var(--line-height-body);
	      font-size: var(--font-size-body);
	      font-weight: var(--font-weight-body);
	      font-family: var(--font-sans);
	    }
@utility text-variant-label1 {
	      line-height: var(--line-height-label1);
	      font-size: var(--font-size-label1);
	      font-weight: var(--font-weight-label1);
	      font-family: var(--font-sans);
	    }
@utility text-variant-label2 {
	      line-height: var(--line-height-label2);
	      font-size: var(--font-size-label2);
	      font-weight: var(--font-weight-label2);
	      font-family: var(--font-sans);
	    }
@utility text-variant-label3 {
	      line-height: var(--line-height-label3);
	      font-size: var(--font-size-label3);
	      font-weight: var(--font-weight-label3);
	      font-family: var(--font-sans);
	    }
@utility text-variant-label4 {
	      line-height: var(--line-height-label4);
	      font-size: var(--font-size-label4);
	      font-weight: var(--font-weight-label4);
	      font-family: var(--font-sans);
	    }
@utility text-variant-legal {
	      line-height: var(--line-height-legal);
	      font-size: var(--font-size-legal);
	      font-weight: var(--font-weight-legal);
	      font-family: var(--font-sans);
	    }"
`;
